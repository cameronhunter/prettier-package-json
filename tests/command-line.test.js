const spawn = require('./__helpers__/spawn');
const path = require('path');

const bin = (command) => path.relative(process.cwd(), path.join(__dirname, '..', 'bin', command));
const fixture = (name) => path.relative(process.cwd(), path.join(__dirname, '__fixtures__', name));

const testCommand = (cmdline, cb) => {
  const [command, ...args] = cmdline.split(' ');

  test(`${command} ${args.join(' ')}`, () => {
    !cb && expect.assertions(1);
    return spawn(bin(command), args).then((result) => {
      if (cb) {
        cb(result);
      } else {
        expect(result).toMatchSnapshot();
      }
    });
  });
};

testCommand(`prettier-package-json ${fixture('package-1.json')}`);
testCommand(`prettier-package-json --use-tabs ${fixture('package-1.json')}`);
testCommand(`prettier-package-json --tab-width 8 ${fixture('package-1.json')}`);
testCommand(`prettier-package-json ${fixture('package-*.json')}`);
testCommand(`prettier-package-json --list-different ${fixture('package-*.json')}`);
testCommand(`prettier-package-json --list-different ${fixture('missing.json')}`);
