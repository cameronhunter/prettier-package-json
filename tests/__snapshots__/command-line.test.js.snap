// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prettier-package-json --list-different tests/__fixtures__/package-*.json 1`] = `
Object {
  "exitCode": 0,
  "stderr": "",
  "stdout": "tests/__fixtures__/package-1.json
",
}
`;

exports[`prettier-package-json --tab-width 8 tests/__fixtures__/package-1.json 1`] = `
Object {
  "exitCode": 0,
  "stderr": "",
  "stdout": "{
        \\"name\\": \\"prettier-package-json\\",
        \\"description\\": \\"Prettier formatter for package.json files\\",
        \\"author\\": \\"Cameron Hunter <hello@cameronhunter.co.uk>\\",
        \\"license\\": \\"MIT\\",
        \\"version\\": \\"1.0.1\\",
        \\"main\\": \\"src/index.js\\",
        \\"bin\\": {
                \\"prettier-package-json\\": \\"./bin/prettier-package-json\\"
        },
        \\"files\\": [
                \\"src\\",
                \\"test\\"
        ],
        \\"scripts\\": {
                \\"precommit\\": \\"./bin/prettier-package-json --write && git add package.json\\",
                \\"test\\": \\"jest\\"
        },
        \\"dependencies\\": {
                \\"commander\\": \\"^2.9.0\\",
                \\"fs-extra\\": \\"^3.0.1\\",
                \\"glob\\": \\"^7.1.2\\",
                \\"sort-object-keys\\": \\"^1.1.2\\"
        },
        \\"devDependencies\\": {
                \\"cross-spawn\\": \\"^5.1.0\\",
                \\"husky\\": \\"^0.13.4\\",
                \\"jest\\": \\"^20.0.4\\"
        },
        \\"keywords\\": [
                \\"formatter\\",
                \\"package.json\\",
                \\"prettier\\",
                \\"sort\\"
        ]
}

",
}
`;

exports[`prettier-package-json --use-tabs tests/__fixtures__/package-1.json 1`] = `
Object {
  "exitCode": 0,
  "stderr": "",
  "stdout": "{
	\\"name\\": \\"prettier-package-json\\",
	\\"description\\": \\"Prettier formatter for package.json files\\",
	\\"author\\": \\"Cameron Hunter <hello@cameronhunter.co.uk>\\",
	\\"license\\": \\"MIT\\",
	\\"version\\": \\"1.0.1\\",
	\\"main\\": \\"src/index.js\\",
	\\"bin\\": {
		\\"prettier-package-json\\": \\"./bin/prettier-package-json\\"
	},
	\\"files\\": [
		\\"src\\",
		\\"test\\"
	],
	\\"scripts\\": {
		\\"precommit\\": \\"./bin/prettier-package-json --write && git add package.json\\",
		\\"test\\": \\"jest\\"
	},
	\\"dependencies\\": {
		\\"commander\\": \\"^2.9.0\\",
		\\"fs-extra\\": \\"^3.0.1\\",
		\\"glob\\": \\"^7.1.2\\",
		\\"sort-object-keys\\": \\"^1.1.2\\"
	},
	\\"devDependencies\\": {
		\\"cross-spawn\\": \\"^5.1.0\\",
		\\"husky\\": \\"^0.13.4\\",
		\\"jest\\": \\"^20.0.4\\"
	},
	\\"keywords\\": [
		\\"formatter\\",
		\\"package.json\\",
		\\"prettier\\",
		\\"sort\\"
	]
}

",
}
`;

exports[`prettier-package-json tests/__fixtures__/package-*.json 1`] = `
Object {
  "exitCode": 0,
  "stderr": "",
  "stdout": "{
  \\"name\\": \\"prettier-package-json\\",
  \\"description\\": \\"Prettier formatter for package.json files\\",
  \\"author\\": \\"Cameron Hunter <hello@cameronhunter.co.uk>\\",
  \\"license\\": \\"MIT\\",
  \\"version\\": \\"1.0.1\\",
  \\"main\\": \\"src/index.js\\",
  \\"bin\\": {
    \\"prettier-package-json\\": \\"./bin/prettier-package-json\\"
  },
  \\"files\\": [
    \\"src\\",
    \\"test\\"
  ],
  \\"scripts\\": {
    \\"precommit\\": \\"./bin/prettier-package-json --write && git add package.json\\",
    \\"test\\": \\"jest\\"
  },
  \\"dependencies\\": {
    \\"commander\\": \\"^2.9.0\\",
    \\"fs-extra\\": \\"^3.0.1\\",
    \\"glob\\": \\"^7.1.2\\",
    \\"sort-object-keys\\": \\"^1.1.2\\"
  },
  \\"devDependencies\\": {
    \\"cross-spawn\\": \\"^5.1.0\\",
    \\"husky\\": \\"^0.13.4\\",
    \\"jest\\": \\"^20.0.4\\"
  },
  \\"keywords\\": [
    \\"formatter\\",
    \\"package.json\\",
    \\"prettier\\",
    \\"sort\\"
  ]
}

{
  \\"name\\": \\"prettier-package-json\\",
  \\"description\\": \\"Prettier formatter for package.json files\\",
  \\"author\\": \\"Cameron Hunter <hello@cameronhunter.co.uk>\\",
  \\"license\\": \\"MIT\\",
  \\"version\\": \\"1.0.1\\",
  \\"main\\": \\"src/index.js\\",
  \\"bin\\": {
    \\"prettier-package-json\\": \\"./bin/prettier-package-json\\"
  },
  \\"scripts\\": {
    \\"precommit\\": \\"./bin/prettier-package-json --write && git add package.json\\",
    \\"test\\": \\"jest\\"
  },
  \\"dependencies\\": {
    \\"commander\\": \\"^2.9.0\\",
    \\"fs-extra\\": \\"^3.0.1\\",
    \\"glob\\": \\"^7.1.2\\",
    \\"sort-object-keys\\": \\"^1.1.2\\"
  },
  \\"devDependencies\\": {
    \\"cross-spawn\\": \\"^5.1.0\\",
    \\"husky\\": \\"^0.13.4\\",
    \\"jest\\": \\"^20.0.4\\"
  },
  \\"keywords\\": [
    \\"formatter\\",
    \\"package.json\\",
    \\"prettier\\",
    \\"sort\\"
  ]
}

",
}
`;

exports[`prettier-package-json tests/__fixtures__/package-1.json 1`] = `
Object {
  "exitCode": 0,
  "stderr": "",
  "stdout": "{
  \\"name\\": \\"prettier-package-json\\",
  \\"description\\": \\"Prettier formatter for package.json files\\",
  \\"author\\": \\"Cameron Hunter <hello@cameronhunter.co.uk>\\",
  \\"license\\": \\"MIT\\",
  \\"version\\": \\"1.0.1\\",
  \\"main\\": \\"src/index.js\\",
  \\"bin\\": {
    \\"prettier-package-json\\": \\"./bin/prettier-package-json\\"
  },
  \\"files\\": [
    \\"src\\",
    \\"test\\"
  ],
  \\"scripts\\": {
    \\"precommit\\": \\"./bin/prettier-package-json --write && git add package.json\\",
    \\"test\\": \\"jest\\"
  },
  \\"dependencies\\": {
    \\"commander\\": \\"^2.9.0\\",
    \\"fs-extra\\": \\"^3.0.1\\",
    \\"glob\\": \\"^7.1.2\\",
    \\"sort-object-keys\\": \\"^1.1.2\\"
  },
  \\"devDependencies\\": {
    \\"cross-spawn\\": \\"^5.1.0\\",
    \\"husky\\": \\"^0.13.4\\",
    \\"jest\\": \\"^20.0.4\\"
  },
  \\"keywords\\": [
    \\"formatter\\",
    \\"package.json\\",
    \\"prettier\\",
    \\"sort\\"
  ]
}

",
}
`;
